doc/glossary.rst

```diff
@@ -127,6 +127,10 @@ Glossary of Terms
       Power gating reduces power consumption by shutting off areas of an
       integrated circuit that are not in use.
 
+   public symbol
+      Any function, structure or macro defined inside the ``include/zephyr``
+      folder that is intended for use by applications or drivers.
```

Comment:
Isn't any function/struct/macro meant to be used by application or drivers?

Regardless, my fear with flagging the whole `include/zephyr` namespace as "public" we'll start talking about more barriers for API changes, some files in there may not be meant to be "public" but are just there so they can used by modules.

Reply:
Some private APIs, structs, and macros exist in `include/zephyr`, but normally we try to hide those from Doxygen. I think generally public API is what is doxygenated.

I would maybe suggest changing the wording here to any "public API"?

Reply:
I suggest we discuss this tomorrow in the Arch WG meeting.